File: app/src/main/java/com/mrgames13/jimdo/feinstaubapp/HelpClasses/TimeFormatter.java
/*
 * Copyright © 2019 Marc Auberer. All rights reserved.
 */

package com.mrgames13.jimdo.feinstaubapp.HelpClasses;

import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.formatter.ValueFormatter;

import java.text.SimpleDateFormat;
import java.util.Date;

public class TimeFormatter extends ValueFormatter {

    // Variables as objects
    private SimpleDateFormat sdf;

    // Variables
    private long first_timestamp;

    public TimeFormatter(long first_timestamp) {
        sdf = new SimpleDateFormat("HH:mm");
        this.first_timestamp = first_timestamp;
    }

    @Override
    public String getAxisLabel(float value, AxisBase axis) {
        Date date = new Date();
        date.setTime((long) (value * 1000 + first_timestamp));
        return sdf.format(date);
    }
}

File: app/src/main/java/com/mrgames13/jimdo/feinstaubapp/HelpClasses/TimeFormatter.kt
/*
 * Copyright © 2019 Marc Auberer. All rights reserved.
 */

package com.mrgames13.jimdo.feinstaubapp.HelpClasses

import com.github.mikephil.charting.components.AxisBase
import com.github.mikephil.charting.formatter.ValueFormatter
import java.text.SimpleDateFormat
import java.util.*

class TimeFormatter(private val first_timestamp: Long) : ValueFormatter() {

    // Variables as objects
    private val sdf: SimpleDateFormat = SimpleDateFormat("HH:mm", Locale.getDefault())

    override fun getAxisLabel(value: Float, axis: AxisBase?): String {
        val date = Date()
        date.time = (value * 1000 + first_timestamp).toLong()
        return sdf.format(date)
    }
}